troposphere/static/js/actions/APITokenActions.js

```diff
@@ -0,0 +1,55 @@
+import APITokenConstants from "constants/APITokenConstants";
+import APIToken from "models/APIToken";
+import NotificationController from "controllers/NotificationController";
+import Utils from "./Utils";
+
+export default {
+    create: ({name, atmo_user}, successCallback, failCallback) => {
+        if (!name) throw new Error("Missing Token name");
+        if (!atmo_user) throw new Error("Missing Token author");
+        let apiToken = new APIToken({
+            name,
+            atmo_user
+        });
+
+        // Add token optimistically
+        Utils.dispatch(APITokenConstants.ADD_TOKEN, {apiToken});
+
+        apiToken
```

Comment:
Change this method so that a promise is returned.

```
let promise = apiToken.save()
promise.done(...)
promise.fail(...)
return promise
```

Reply:
This does prove that the model is indeed a promise and removes the `then is not a function` error I was getting before. `then` however isn't getting called... I'll keep digging.

Reply:
The model is not a promise

Reply:
All I can think of is...

https://www.youtube.com/watch?v=9iSlPoQm2XY